lib/datadog/core/configuration/config_helper.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require_relative 'assets/supported_configurations'
+require_relative '../logger'
+require 'json'
+
+module Datadog
+  module Core
+    module Configuration
+      module ConfigHelper
+        # Load the supported configurations from JSON file. We will do this AOT.
+        SUPPORTED_CONFIGURATIONS = Assets::SUPPORTED_CONFIG_DATA[:supportedConfigurations] || {}
+        private_constant :SUPPORTED_CONFIGURATIONS
+
+        ALIASES = Assets::SUPPORTED_CONFIG_DATA[:aliases] || {}
+        private_constant :ALIASES
+
+        DEPRECATIONS = Assets::SUPPORTED_CONFIG_DATA[:deprecations] || {}
+        private_constant :DEPRECATIONS
+
+        ALIAS_TO_CANONICAL = Assets::ALIAS_TO_CANONICAL || {}
+        private_constant :ALIAS_TO_CANONICAL
+
+        # Returns the environment variable, if it's supported or a non Datadog
+        # configuration. Otherwise, it throws an error.
+        #
+        # @param name [String] Environment variable name
+        # @return [String, nil] The environment variable value
+        # @raise [RuntimeError] if the configuration is not supported
+        def get_environment_variable(name, env_vars: ENV, source: 'environment')
+          # Log deprecations once for environment, fleet config file and local config file.
+          log_deprecations(env_vars, source)
+
+          # datadog-ci-rb is using dd-trace-rb config DSL, which uses this method.
+          # Until we've correctly implemented support for datadog-ci-rb, we disable config inversion if ci is enabled.
+          if !defined?(::Datadog::CI) &&
+              (name.start_with?('DD_', 'OTEL_') || ALIAS_TO_CANONICAL[name]) &&
+              !SUPPORTED_CONFIGURATIONS[name]
+            if defined?(@dd_test_environment) && @dd_test_environment
+              if ALIAS_TO_CANONICAL[name]
+                raise "Please use #{ALIAS_TO_CANONICAL[name]} instead of #{name}."
+              else
+                raise "Missing #{name} env/configuration in \"supported-configurations.json\" file."
+              end
+            end
+            # TODO: Send telemetry to know if we ever miss an env var
+            return nil
+          end
+
+          config = env_vars[name]
+          if config.nil? && ALIASES[name]
+            ALIASES[name].each do |alias_name|
+              return env_vars[alias_name] if env_vars[alias_name]
+            end
+          end
+
+          config
+        end
+
+        private
+
+        # An env can be deprecated without a replacement (e.g.: if we remove a feature)
```

Comment:
I think this comment should be deleted - it's pretty obvious, and it's not the right place for it